File: java-demo/src/main/java/zlc/season/rxdownload/java_demo/BaseApplication.java
package zlc.season.rxdownload.java_demo;


import android.app.Application;

import zlc.season.rxdownload3.core.DownloadConfig;
import zlc.season.rxdownload3.extension.ApkInstallExtension;
import zlc.season.rxdownload3.extension.ApkOpenExtension;

public class BaseApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        DownloadConfig.Builder builder = DownloadConfig.Builder.Companion.create(this)
                .enableDb(true)
                .setDbActor(new CustomSqliteActor(this))
                .enableService(true)
                .enableNotification(true)
                .addExtension(ApkInstallExtension.class)
                .addExtension(ApkOpenExtension.class);

        DownloadConfig.INSTANCE.init(builder);
    }
}

File: kotlin-demo/src/main/java/zlc/season/rxdownload/kotlin_demo/BaseApplication.kt
package zlc.season.rxdownload.kotlin_demo

import android.app.Application
import android.os.Environment
import zlc.season.rxdownload3.core.DownloadConfig
import zlc.season.rxdownload3.extension.ApkInstallExtension
import zlc.season.rxdownload3.extension.ApkOpenExtension


class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        val builder = DownloadConfig.Builder.create(this)
                .setDebug(true)
                .enableNotification(true)
                .addExtension(ApkInstallExtension::class.java)
                .addExtension(ApkOpenExtension::class.java)
                .setDefaultPath(Environment.getExternalStorageDirectory().toString() + "/Huawei" + "/Themes")

        DownloadConfig.init(builder)
    }
}